### PR TITLE
BREAKING(bytes): deprecate `concat()` signatures that don't use `Uint8Array[]` argument

### PR DESCRIPTION
**Pull request by iuioiua**
This change deprecates the `concat()` signatures that use spread and depth > 1 arguments in favour of a single argument of an array of `Uint8Array`s. This will enable the function to remain simple and performant while still taking care of scenarios where large numbers of `Uint8Array`s need to be concatenated.

This comes from work done in #3698.

### CODE REVIEW

**Review by lino-levan — APPROVED** [review]
Would be a good time to aim for 100% coverage in concat as a bonus for this PR, but this LGTM.

File: bytes/concat.ts [hunk 1]
```diff
@@ -1,18 +1,57 @@
...
+ *
+ * @example
+ * ```ts
+ * import { concat } from "https://deno.land/std@$STD_VERSION/bytes/concat.ts";
+ *
+ * const a = new Uint8Array([0, 1, 2]);
+ * const b = new Uint8Array([3, 4, 5]);
+ * concat(a, b); // Uint8Array(6) [ 0, 1, 2, 3, 4, 5 ]
+ * ```
+ */
```
Comment (kt3k) [inline bytes/concat.ts:47]: I don't think we need to have this jsdoc. This is for implementatioin signature ([ref](https://www.typescriptlang.org/docs/handbook/2/functions.html#overload-signatures-and-the-implementation-signature)) and that is not visible to the users.

Comment (iuioiua) [reply bytes/concat.ts]: Ah, yes, you did say that. Fixed.

**Review by kt3k — APPROVED** [review]
LGTM

**Pull request merged**